.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: osgood CI suite
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+    - name: Install Pre-Commit dependencies
+      run: pip install scspell3k
```

Comment:
Use typos

Reply:
I would actually have the CI check that `pre-commit` passes and add the `typos` to the pre-commit of the project. Since this is a non-conda project, use the pre-commit config from the `typos` repo.